### PR TITLE
JS: model `path-is-inside`+`is-path-inside` for `js/path-injection`

### PR DESCRIPTION
**Pull request by esbena**
The following pattern sanitizes paths wrt. path injection.

```js
const isPathInside = require('path-is-inside');

if (!isPathInside(file, process.cwd())) {
    this.body = 'Bad Request';
    this.status = 400;
}
```

This PR models two confusingly similar packages: 'path-is-inside' and 'is-'path-inside' for that purpose.

Evaluation is pending.

### CODE REVIEW

**Comment by erik-krogh** [comment]
```JavaScript
var inside = require("path-is-inside")
inside("foo/bar/../../../../../etc/passwd", "foo/bar") === true
```

The `is-path-inside` library actually works, as it uses `path.resolve(..)` to "compress" the paths.

File: javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll [hunk 1]
```diff
@@ -355,6 +355,33 @@ module TaintedPath {
...
+        check = this
+      )
+    }
+
+    override predicate blocks(boolean outcome, Expr e) {
+      e = checked.asExpr() and
+      outcome = true
+      // Note that `is-path-inside` states:
+      // > "Note that relative paths are resolved against process.cwd() to make them absolute."
+      // so it may be possible to undo this sanitization by later prepending some other path to the sanitized path.
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll]: Did you mean to leave this comment here? Doesn't seem like something we need to worry about.

File: javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll [hunk 2]
```diff
@@ -355,6 +355,33 @@ module TaintedPath {
...
+  /**
+   * An expression of form `isInside(x, y)` or similar, where `isInside` is
+   * a library check for the relation between `x` and `y`.
+   */
+  class IsInsideCheckSanitizer extends DataFlow::BarrierGuardNode {
+    DataFlow::Node checked;
+
+    IsInsideCheckSanitizer() {
+      exists(string name, DataFlow::CallNode check |
+        name = "path-is-inside" or
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll]: In line with @erik-krogh's comment, could you make `path-is-inside` only sanitize normalized absolute paths?

Comment (esbena) [reply javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll]: I am waiting to see if Erik's PR against `path-is-inside` gets accepted.

Comment (esbena) [reply javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll]: That PR seems to have entered the void. `path-is-inside` is now a weaker sanitizer.

**Comment by esbena** [comment]
I forgot about `--dpm`, but the [evaluation](https://git.semmle.com/esben/dist-compare-reports/tree/js/support-path-is-inside_1582615411820) seems uneventful. I believe this can be merged now.

**Pull request merged**